flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/SplitAssignmentStatus.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.flink.connector.kinesis.source.enumerator;
+
+import org.apache.flink.annotation.Internal;
+
+/**
+ * Assignment status of {@link org.apache.flink.connector.kinesis.source.split.KinesisShardSplit}.
+ */
+@Internal
+public enum SplitAssignmentStatus {
+    ASSIGNED(0),
+    UNASSIGNED(1);
+
+    private final int statusCode;
+
+    SplitAssignmentStatus(int statusCode) {
+        this.statusCode = statusCode;
+    }
+
+    public int getStatusCode() {
+        return statusCode;
+    }
+
+    public static SplitAssignmentStatus fromStatusCode(int statusCode) {
+        for (SplitAssignmentStatus status : SplitAssignmentStatus.values()) {
+            if (status.getStatusCode() == statusCode) {
+                return status;
+            }
+        }
+
+        throw new IllegalArgumentException("Unknown status code: " + statusCode);
+    }
```

Comment:
Noted that we have set up the serialization arrangement for enums correctly so that we can add more enums safely in the future :)